Time And Time Again/src/TimeAndTimeAgain.java

```diff
@@ -0,0 +1,59 @@
+import java.util.Scanner;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+
+public class TimeAndTimeAgain
+	{
+
+		public static void main(String[] args)
+			{
+				try(Scanner input = new Scanner(System.in))
+				{
+					int numTimeCases = input.nextInt();
+					
+					//gets all the digits before the s and stops at s
+					Pattern hourRegex = Pattern.compile(".*(\\d+)h.*");
```

Comment:
With this regex, you're missing a ? symbol that will cause an issue with reading in the text. It should look like `Pattern.compile(".*?(\\d+)s.*");`

The `.*?` part is looking for anything zero or more times, stopping when it first finds a digit defined with `(\\d+)`. The ? is a "reluctant" qualifier which means it stops as soon as it can. Versus, if you don't use the ?, it's a "greedy" statement which means it actually goes past that first digit in your number. So if your input was "32h" it would only give you "2" back, if it was "5m" it would give you no number back. 

Sort of a complicated reasoning ... so just make sure you put a ? in your Pattern for hour, minute, and second!